articles/moricho.re

```diff
@@ -0,0 +1,71 @@
+= Goによるコンテナランタイム自作入門
+
+== はじめに
+こんにちは、@_moricho_@<fn>{_moricho_}です。
+〜〜〜
+
+//footnote[_moricho_][@<href>{https://twitter.com/_moricho_}]
+
+== コンテナランタイムとは？
+コンテナランタイムはコンテナの実行管理をするものです。コンテナランタイムは大きく分けて「ハイレベルコンテナランタイム」と「ローレベルランタイム」に分割されます。
+例えばDockerなどのコンテナを扱う際は、基本的に「ハイレベルランタイム => ローレベルランタイム」の順で処理が流れていきます。
+
+1. ハイレベルランタイム
+ハイレベルランタイムでは直接コンテナを操作するわけではなく、その次のローレベルランタイムへ命令を渡す役割を果たします。
+またここではイメージの管理(pull/push/rm…)も担います。
+
+2.  ローレベルランタイム
+ハイレベルランタイムから受け取った指示でコンテナの起動や停止をキックするなど、コンテナの直接的な操作を担当する部分です。
+今回は、コンテナのコアとなるこの部分を作っていきます。
+
+
+== コンテナの概要
+
+ここでは、コンテナをコンテナたらしめている仕組みをざっくり説明します。
+実際にコンテナランタイムを実装していく上で見通しをよくするためにも必要になる知識ですので、ざっと理解しておきましょう。
+
+=== コンテナとは？
+コンテナを一言で言うとなんでしょうか？コンテナとは「ホストOSのリソースを隔離・制限したプロセス」と言えます。
+プロセスであるため、通常のアプリケーションと遜色ないレベルで素早く実行できます。また、ホストOSとリソースを共有しているのがポイントです。
+
+=== なぜ異なるOSベースのイメージが動くの？
```

Comment:
**[textlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Disallow to use "？". (ja-technical-writing/no-exclamation-question-mark)